src/metabase/models/emitter.clj

```diff
@@ -27,14 +41,15 @@
          {:primary-key (constantly :emitter_id)}))
 
 (defn- pre-insert
-  [emitter]
+  [{action-id :action_id, :as emitter}]
+  {:pre [(integer? action-id)]}
   (let [base-emitter (db/insert! Emitter (select-keys emitter [:parameter_mappings :options]))]
     (db/insert! EmitterAction (-> emitter
                                   (select-keys [:action_id])
                                   (assoc :emitter_id (:id base-emitter))))
     (-> emitter
         (select-keys [:dashboard_id :card_id])
-        (assoc :emitter_id (:id base-emitter)))))
+        (assoc :emitter_id (u/the-id base-emitter)))))
```

Comment:
Style question: Is it preferable to always use u/the-id vs :id for this usecase?